Реализовать UI слой экрана Выбор отрасли

JudjinGM opened the pull request.

JudjinGM committed:
Реализовал получение регионов по приходчщим аргументам(id страны). Если id == null, то приходит весь списко регионов. Сделал филтрацию по имени региона.

Навигация в этой ветке неверная, сделал для возможности перейти сразу на выбор регинов.

JudjinGM committed:
Поправил замечания:
1. Клик на весь элемент
2. Навигация и передача area через fragment result api.

JudjinGM committed:
В процессе написания экрана отрасли.

JudjinGM committed:
Решение проблемы с адаптером

JudjinGM committed:
1. Реализовал получение списка всех отраслей
2. Реализовал поиск по списку всех отраслей
3. Реализовал выбор единственной отрасли из списка всех отраслей и передачу во fragment result после нажатия кнопки выбрать.

JudjinGM committed:
Changes for merge

JudjinGM committed:
Merge branch 'dev' into feature/FLTR-234/industry_search

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/common/domain/model/filter_models/AreaFilter.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/di/FilterUiModule.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/AreaCountryViewHolder.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/fragment/FilteringSettingsFragment.kt
#	app/src/main/java/ru/practicum/android/diploma/filter/ui/viewModel/FilteringSettingsViewModel.kt

manualcheg commented on app/src/main/java/ru/practicum/android/diploma/common/domain/model/filter_models/AreaFilter.kt:
```diff
@@ -1,7 +1,36 @@
 
+import android.os.Parcel
+import android.os.Parcelable
+
```
Пишут, что можно не переопределять методы, а использовать аннотацию @Parcelize. 
Будет использована кодогенерация. https://developer.android.com/kotlin/parcelize

JudjinGM committed:
Changes for merge #2.
Исправил взаимодействие между фрагментами, поправил логику в настройках фильтров. Немного подчистил проект.
